Bottom: 5dddba59928d161092ffe9d50d5b725996edaffc
Top:    4024e8fd7bc38e6d32f208f1f3e8f3176dd7a93b
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-27 05:07:33 +0100

Play with grid snapping heuristics a little

Require the pointer to be inside the bounding box of a pin / pad / via before snapping.
This makse it MUCH easier to place vias in a regular grid spaced at the PCB grid spacing.


---

diff --git a/src/crosshair.c b/src/crosshair.c
index 25c77cc..09f94dc 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -1158,7 +1158,7 @@ FitCrosshairIntoGrid (Coord X, Coord Y)
   ans = NO_TYPE;
   if (PCB->RatDraw || TEST_FLAG (SNAPPINFLAG, PCB))
     ans = SearchObjectByLocation (PAD_TYPE, &ptr1, &ptr2, &ptr3,
-                                  Crosshair.X, Crosshair.Y, PCB->Grid / 2);
+                                  Crosshair.X, Crosshair.Y, 0);
 
   /* Avoid self-snapping when moving */
   if (ans != NO_TYPE &&
@@ -1215,7 +1215,7 @@ FitCrosshairIntoGrid (Coord X, Coord Y)
   ans = NO_TYPE;
   if (PCB->RatDraw || TEST_FLAG (SNAPPINFLAG, PCB))
     ans = SearchObjectByLocation (PIN_TYPE, &ptr1, &ptr2, &ptr3,
-                                  Crosshair.X, Crosshair.Y, PCB->Grid / 2);
+                                  Crosshair.X, Crosshair.Y, 0);
 
   /* Avoid self-snapping when moving */
   if (ans != NO_TYPE &&
@@ -1233,7 +1233,7 @@ FitCrosshairIntoGrid (Coord X, Coord Y)
   ans = NO_TYPE;
   if (TEST_FLAG (SNAPPINFLAG, PCB))
     ans = SearchObjectByLocation (VIA_TYPE, &ptr1, &ptr2, &ptr3,
-                                  Crosshair.X, Crosshair.Y, PCB->Grid / 2);
+                                  Crosshair.X, Crosshair.Y, 0);
 
   /* Avoid snapping vias to any other vias */
   if (Settings.Mode == MOVE_MODE &&
